### PR TITLE
using non recursive bfs to find the shortest path

### PR DESCRIPTION
**Pull request by ErfanGerami**

### CODE REVIEW

File: graph.py [hunk 1]
```diff
@@ -19,24 +21,77 @@ def DeleteEdge(self,node1,node2):
     #-------------------------------------
     def IsThere(self,node):
         '''return True if the node is in the graph and False if it is not'''
-        if node in self.network:
-            return True
-        return False
+        return (node in self.network)
```
Comment (mamadmr) [inline graph.py:24]: why don't you use mark instead? 
or a concept like this ? 
becuase searcnig like this is so slow

File: graph.py [hunk 2]
```diff
@@ -19,24 +21,77 @@ def DeleteEdge(self,node1,node2):
...
         '''returns a set that contains the nodes that the given node is connected to'''
         return self.network[node]
-                
+    
+    #---------------------------------------------
+    def StartMarking(self):
+        '''use to reset or start Marking'''
+        maximum_node=max(self.network)
+        self.marks=[]
+        for i in range(0,maximum_node+1):
```
Comment (mamadmr) [inline graph.py:34]: [0 for i in range(max_node+1)

Comment (mamadmr) [reply graph.py:34]: don't use max_node, for example they gives you the request for go from node 1 to node 50 but the user just has max_node 20 so you will get runtime error

File: graph.py [hunk 3]
```diff
@@ -19,24 +21,77 @@ def DeleteEdge(self,node1,node2):
...
+    def GetMark(self,node):
+        return self.marks[node]
+    
+    def FindPath(self,start,dest):
+        '''finds a path betwean start and des and returns it as a list'''
+        marks_temp=self.marks.copy()
+        self.StartMarking()
+
+        path=self._FindPath(start,dest)
+        self.marks=marks_temp.copy()
```
Comment (mamadmr) [inline graph.py:49]: why do you try to save last marks ?

Comment (ErfanGerami) [reply graph.py:49]: Because the marks are not just for me .assume that someone else wants to use it .

File: graph.py (hunk 2 above)
Comment (ErfanGerami) [reply graph.py:34]: I didnt understand?its just for having all marks  of all nodes in a list and the max marks we have equals to the max node .

**Pull request closed without merging**